garak/attempt.py

```diff
@@ -105,6 +105,24 @@ def as_dict(self) -> dict:
             "messages": self.messages,
         }
 
+    @classmethod
+    def from_dict(cls, dicti):
```

Comment:
-- out of scope for here, but we should implement serialization/deserialization for `Attempt`s